### PR TITLE
Doc updates

### LINKED ISSUE #71: Document how to Register for NetApp files
Currently we have the HA option that allows users to use the NetApp system items in Azure; however, one must have their subscription registered to use this extra feature given it generates cost for the system.

Here is the reference link from MS:

https://docs.microsoft.com/en-us/azure/azure-netapp-files/azure-netapp-files-register?WT.mc_id=Portal-Microsoft_Azure_NetApp

### PR DESCRIPTION
**Pull request by manoatsas**
- Add IaC description in README.md
- close #71 
- Update Docker usage docs, simplify TerraformAzureAuthentication

### CODE REVIEW

File: docs/user/AzureHelpTopics.md [hunk 1]
```diff
@@ -60,18 +60,25 @@ If in case `$SP_PASSWD` value is lost or has expired, you can reset it with this
 az ad sp credential reset --name http://$USER
 ```
 
-You can find more details on Azure documentation on how to `create/retrieve/manage/reset` [Azure Service Principal with AzureCLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest)
+If you don't have an Azure account with **Owner** role please check with your Azure account administrator. You can find more details on Azure documentation on how to `create/retrieve/manage/reset` [Azure Service Principal with AzureCLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest). 
+
+To learn more about how Azure Role based access control works refer to the docs:
```
Comment (enderm) [inline docs/user/AzureHelpTopics.md:65]: capitalize " Azure Role Based Access Control"

File: docs/user/AzureHelpTopics.md [hunk 2]
```diff
@@ -91,7 +98,7 @@ az identity create --resource-group <my resource group> --name <my identity name
 PID=$(az identity show --resource-group <my resource group> --name <my identity name> --output tsv --query principalId)
 az role assignment create --assignee $PID --role Contributor
 ```
-## Assign the Managed Identity to a VM
+## How to assign User-assigned Managed Identity to a VM
```
Comment (enderm) [inline docs/user/AzureHelpTopics.md:101]: add article: How to assign  _a_ User-assigned...

File: docs/user/DockerUsage.md [hunk 3]
```diff
@@ -2,34 +2,56 @@
...
 
 Add volume mounts to the `docker run` command for all files and directories that must be accessible from inside the container.
 
-The most common file reference is the value of the [`ssh_public_key`](./CONFIG-VARS.md#required-variables) variable in the `terraform.tfvars` file.
+The most common file reference is the value of the [`ssh_public_key`](../CONFIG-VARS.md#required-variables) variable in the `terraform.tfvars` file.
 
 Note that local references to `$HOME` (or "`~`") need to map to the root directory `/` in the container.
 
+You also need to mount the local directory containing your `terraform.tfvars`, that can be your current directory e.g., $(pwd). This is the same local directory where Terraform state file `terraform.tfstate` will be written. To grant Docker permission to write to the directory use [Docker `--group-add root` option](https://docs.docker.com/engine/reference/run/#additional-groups) e.g.
+
```
Comment (enderm) [inline docs/user/DockerUsage.md:41]: I was acctually wondering about that. Isn' t it the --user option that allows the container to read/write to the local directory?
Not sure what "--group-add root" is there for, but not for write access to $pwd

**Review by enderm — CHANGES_REQUESTED** [review]
love it! Only minor quibs.

File: docs/user/DockerUsage.md (hunk 3 above)
Comment (manoatsas) [reply docs/user/DockerUsage.md:41]: Yes I should remove that, it's to override `--user` option and to run as root

File: docs/user/AzureHelpTopics.md (hunk 2 above)
Comment (manoatsas) [reply docs/user/AzureHelpTopics.md:101]: can you elaborate?

Comment (enderm) [reply docs/user/AzureHelpTopics.md:101]: add the word "a" between "assign" and "User-assigned" (I think the grammatical term for that is "indefinite article" :) )

File: docs/user/TerraformAzureAuthentication.md [hunk 4]
```diff
@@ -18,8 +18,8 @@ To authenticate Terraform to access Azure, you will need to set the following fo
 | :--- | ---: | ---: | ---: |
 | tenant_id | your Azure tenant id | string  |
```
Comment (thpang) [inline docs/user/TerraformAzureAuthentication.md:19]: The tenant_id and subscription_id items seem to be missing a column definition.

File: docs/CONFIG-VARS.md [hunk 5]
```diff
@@ -41,8 +41,8 @@ Find details on how to retrieve that information under [Azure Help Topics](./use
 
 | Name | Description | Type | Default |
 | :--- | ---: | ---: | ---: |
-| tenant_id | your Azure tenant id | string  |
-| subscription_id | your Azure subscription id | string  |
+| tenant_id | your Azure tenant id | string  | |
```
Comment (thpang) [inline docs/CONFIG-VARS.md:44]: So for both tenant_id and subscription_id no default? Just asking as client_id and client_secret seem to have values of ""

File: docs/user/TerraformAzureAuthentication.md (hunk 4 above)
Comment (manoatsas) [reply docs/user/TerraformAzureAuthentication.md]: It should be handled by the formatting row, but fixed it.

File: docs/CONFIG-VARS.md (hunk 5 above)
Comment (manoatsas) [reply docs/CONFIG-VARS.md:44]: Yes, using defaults from https://github.com/sassoftware/viya4-iac-azure/blob/main/variables.tf#L1

Comment (thpang) [reply docs/CONFIG-VARS.md:44]: got it. thx.

**Comment by thpang** [comment]
LGTM

**Pull request merged**